src/Xamarin.Android.Build.Tasks/Tasks/CreateLibraryResourceArchive.cs

```diff
@@ -101,6 +103,27 @@ public override bool Execute ()
 			}
 			return true;
 		}
+
+		void FixApplicationIdInManifests (string directory)
+		{
+			var broken = new List<string> ();
+			string goodFile = null;
+			foreach (var file in Directory.EnumerateFiles (directory, "AndroidManifest.xml", SearchOption.AllDirectories)) {
+				string text = File.ReadAllText (file);
+				if (text.Contains (("dollar_openBracket_applicationId_closeBracket"))) {
+					broken.Add (file);
+				} else {
+					goodFile = file;
+				}
+			}
+			if (!string.IsNullOrEmpty (goodFile)) {
+				foreach (var file in broken) {
+					DateTime lastWriteTime = File.GetLastWriteTimeUtc (file);
+					File.Copy (goodFile, file, overwrite: true);
```

Comment:
Let's see if I understand the scenario properly: the `.aar` file is expected to contain *multiple* `AndroidManifest.xml` files, one of which is "good" and one or more of which is "bad," and we simply replace the "bad" ones with the "good" one.

Is this reasonable? Are we certain that there won't be *other* differences between these files? (I vaguely recall bug reports in which there were multiple `AndroidManifest.xml` files within a `.aar`, and they *weren't* identical, but perhaps I'm mistaken?)

I also don't understand why this needs to be done, given the change in `ManifestDocument.cs`.

Finally, assuming this change is required, it would in turn require that binding projects be *rebuilt* to get this change. Is that practical?